crates/levm/src/vm.rs

```diff
@@ -304,6 +323,68 @@ impl VM {
                         .stack
                         .push(U256::from(current_call_frame.pc - 1));
                 }
+                Opcode::BLOCKHASH => {
+                    let block_number = current_call_frame.stack.pop().unwrap();
+
+                    // If number is not in the valid range (last 256 blocks), return zero.
+                    if block_number
+                        < block_env
+                            .number
+                            .saturating_sub(U256::from(LAST_AVAILABLE_BLOCK_LIMIT))
+                        || block_number >= block_env.number
+                    {
+                        current_call_frame.stack.push(U256::zero());
+                        continue;
+                    }
```

Comment:
Let's try running the tests to remove this condition to ensure we cover these cases in the below `if`. If they pass, let's add a comment in the `else` below explaining that it covers the cases where the block is not in a valid range (last 256 blocks).

Reply:
Ok! Let's keep it then.